contracts/manager/BaseManager.sol

```diff
@@ -155,12 +245,59 @@ contract BaseManager {
     }
 
     /**
-     * ADAPTER ONLY: Interact with a module registered on the SetToken.
+     * MUTUAL UPGRADE**: Authorizes an adapter for a protected module. Operator and Methodologist must
+     * each call this function to execute the update.
+     *
+     * @param _module           Module to authorize adapter for
+     * @param _adapter          Adapter to authorize for module
+     */
+    function authorizeAdapter(address _module, address _adapter)
+        external
+        mutualUpgrade(operator, methodologist)
+    {
+        require(protectedModules[_module].isProtected, "Module not protected");
+        require(isAdapter[_adapter], "Adapter does not exist");
+        require(!protectedModules[_module].authorizedAdapters[_adapter], "Adapter already authorized");
+
+        protectedModules[_module].authorizedAdapters[_adapter] = true;
+        protectedModules[_module].authorizedAdaptersList.push(_adapter);
+
+        emit AdapterAuthorized(_module, _adapter);
+    }
+
+    /**
+     * MUTUAL UPGRADE**: Revokes adapter authorization for a protected module. Operator and Methodologist
+     * must each call this function to execute the update.
```

Comment:
Would note here that in order to remove the extension completely from the contract `removeAdapter` must be called by the `operator`

Reply:
1392513